google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java

```diff
@@ -423,12 +441,37 @@ public Builder setMd5(String md5) {
       return this;
     }
 
+    public Builder setMd5FromHexString(String md5HexString) {
+      if (md5HexString == null) {
+        return this;
+      }
+      if (md5HexString.startsWith("0x")) {
+        md5HexString = md5HexString.replaceFirst("0x", "");
+      }
```

Comment:
Remove code that removes "0x" from the hex string.

Reply:
Removed code that removes 0x from hex string and code that adds prepending 0x. OmitPadding removed